### PR TITLE
Inquirer types

### LINKED ISSUE #4881: Improve typing in oracle methods
After #4834 we have the `query_current_price` https://github.com/prettyirrelevant/rotki/blob/84d95d6062b99be215d28f7fddb75e2c0246e34e/rotkehlchen/externalapis/coingecko.py#L518 getting as argument asset. Due this type we have to resolve the assets but this can be costly.

To improve the situation here we should create two version of the function accepting as type `AssetWithOracles` so we avoid the resolution of the assets again if it has already be done

### PR DESCRIPTION
**Pull request by nebolax**
Closes #4881 

## Checklist

- [x] Change how we work with manual latest price oracle and resolve assets beforehand where possible
- [x] Not related to the issue, but also changed some names (changed variables with name `asset` and type `EvmToken`
 to name `token`)

### CODE REVIEW

File: rotkehlchen/globaldb/manual_price_oracles.py [hunk 1]
```diff
@@ -51,14 +50,7 @@ def query_historical_price(
...
 
-class ManualCurrentOracle(CurrentPriceOracleInterface):
-
-    def __init__(self) -> None:
-        super().__init__(oracle_name='manual current price oracle')
-
-    def rate_limited_in_last(self, seconds: Optional[int] = None) -> bool:
-        return False
-
+class ManualCurrentOracle:
```
Comment (LefterisJP) [inline rotkehlchen/globaldb/manual_price_oracles.py:53]: Why not do what we discussed in discord?



This way you will have to use `Union` for types everywhere while with inheritance you can use the base type.

Sure, at some point you will have to do a type: ignore for the function signature not agreeing with supertype for the ManualCurrentOracle but that's fine, no?

**Review by LefterisJP — COMMENTED** [review]
A question for the approach. And 3 tests failed.

File: rotkehlchen/inquirer.py [hunk 2]
```diff
@@ -410,12 +410,18 @@ def _query_oracle_instances(
         ), (
             'Inquirer should never be called before the setting the oracles'
         )
-        oracles = instance._oracles
-        oracle_instances = instance._oracle_instances
-
-        if skip_onchain:
-            oracles = instance._oracles_not_onchain
-            oracle_instances = instance._oracle_instances_not_onchain
+        if from_asset.is_asset_with_oracles() is True:
```
Comment (LefterisJP) [inline rotkehlchen/inquirer.py:413]: Hmmm why did you change this like that? Shouldn't it be the opposite?

Comment (nebolax) [reply rotkehlchen/inquirer.py:413]: I changed the order of the if-else to avoid doing having to explicitly set oracle_instancee' type

Comment (nebolax) [reply rotkehlchen/inquirer.py:413]: It's fine

**Review by LefterisJP — APPROVED** [review]
LGTM. Frontend tests are flaky
